java/src/com/ibm/streamsx/rest/StreamsConnection.java

```diff
@@ -6,114 +6,35 @@
-
-    private final String userName;
+    private String userName;
+    private String authToken;
     private String url;
-    protected String apiKey;
-    private boolean allowInsecureHosts = false;
-
-    protected Executor executor;
 
-    /**
-     * Connection to IBM Streams
-     * 
-     * @param userName
-     *            String representing the userName to connect to the instance
-     * @param authToken
-     *            String representing the password to connect to the instance
-     * @param url
-     *            String representing the root url to the REST API, for example:
-     *            https:server:port/streams/rest
-     */
-    protected StreamsConnection(String userName, String authToken, String url) {
-        this.userName = userName;
-        String apiCredentials = userName + ":" + authToken;
-        apiKey = "Basic " + DatatypeConverter.printBase64Binary(apiCredentials.getBytes(StandardCharsets.UTF_8));
-
-        executor = Executor.newInstance();
-        setStreamsRESTURL(url);
-    }
-
-    /**
-     * sets the REST API url for this connection removing the trailing slash
-     * 
-     * @param url
-     *            String representing the root url to the REST API, for example:
-     *            https:server:port/streams/rest/resources
-     *            https:server/streams/rest Need to remove the '/resources' for
-     *            all other access to be efficient
-     */
-    protected void setStreamsRESTURL(String iUrl) {
-        // set this originally to what comes in to force a 404 if it's not what
-        // we wanted
-        this.url = iUrl;
-        if (!iUrl.isEmpty()) {
-            // strip trailing slash
-            if (iUrl.charAt(iUrl.length() - 1) == '/') {
-                iUrl = iUrl.substring(0, iUrl.length() - 1);
-            }
-
-            // IBM Streams returns the REST URI Root as
-            // https:server:port/streams/rest/resources
-            // whereas Streaming Analytics returns the REST URI as
-            // https:server/streams/rest
-            // https:server/streams/rest is the common path between them
-            String keyword = "streams/rest";
-            String endString = "";
-            if (getClass() == StreamsConnection.class) {
-                endString = "/resources";
-            }
-
-            int index = iUrl.indexOf(keyword);
-            if (index != -1) {
-                String remants = "";
-                if (index < iUrl.length()) {
-                    remants = iUrl.substring(index + keyword.length(), iUrl.length());
-                }
-                // verify that there's nothing at the end
-                if (remants.equals(endString)) {
-                    // make sure we stop at streams/rest
-                    this.url = iUrl.substring(0, index + keyword.length());
-                }
-            }
-        }
+    protected StreamsConnection(StreamsConnectionInterface delegate,
```

Comment:
Since the existing `StreamsConnection` constructor was protected, it is part of the public api so it needs to remain.

Maybe make this a new constructor and have it package protected.

Reply:
The constructor won't be a problem, but there are some other protected member fields that while I could leave in, almost certainly won't work as they used to, eg. the `apiKey` won't be usable in the same way it used to be (it's changed in the new implementation because it used to be the `Bearer` string for the `Authorization` header, and V2 doesn't use that).

I can leave the members in, but they might not make sense if someone has actually created a derived class that depends on them.